node/src/components/contract_runtime.rs

```diff
@@ -472,6 +474,7 @@ where
                             let correlation_id = CorrelationId::new();
                             let start = Instant::now();
                             let result = engine_state.commit_step(correlation_id, step_request);
```

Comment:
Why not put it inside the `commit*` functions? Those are the ones that "flush" changes to the disk.

Reply:
This may reflect the benchmarks Dan has currently implemented.  The last time I spoke he was actually instrumenting the loop that was extracted [here][1], which is slated for the 1.4.0 release. If he was flushing only after executing the whole block, this is the place in the code as-is to do so.

But it's straightforward to use benchmarks to figure out what to do here:

  - Put the flush after executing the entire block
  - Put the flush after every `commit`/`commit_step`

Reply:
You're right, we should definitely benchmark the alternative.

Reply:
Here are the results of flushing every commit vs the baseline (current functionality):
![image](https://user-images.githubusercontent.com/1497784/125837481-a5fe2cab-1c25-4c94-a6a5-4f070c9ee302.png)

Reply:
Thanks for this @dwerner ! That's a great analysis. Case closed :)